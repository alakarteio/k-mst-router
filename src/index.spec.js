/* eslint-env jest */

import Router from './index'

describe('k-mst-router', () => {
  it('should init with a simple router', () => {
    const router = Router.create({
      screens: [
        {
          name: 'home',
          path: '/',
        },
      ],
    })

    expect(router.toJSON()).toMatchSnapshot()
  })

  it('should init with siblings routes', () => {
    const router = Router.create({
      screens: [
        {
          name: 'home',
          path: '/',
        },
        {
          name: 'login',
          path: '/login',
        },
        {
          name: 'list',
          path: '/list',
        },
      ],
    })

    expect(router.toJSON()).toMatchSnapshot()
  })

  const create = () => Router.create({
    screens: [
      {
        name: 'home',
        path: '/',
      },
      {
        name: 'list',
        path: '/list',
      },
      {
        name: 'edit',
        path: '/list/:id',
      },
    ],
  })

  it('should go to edit sceen', () => {
    // mock
    jsdom.reconfigure({
      url: 'http://monsite.com/list/2',
    })
    window.history.pushState = jest.fn()

    // exec
    const router = create()

    // asserts
    expect({
      router: router.toJSON(),
      pushState: window.history.pushState.mock.calls,
    }).toMatchSnapshot()
  })
})
