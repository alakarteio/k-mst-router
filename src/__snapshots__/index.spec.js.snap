// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`k-mst-router should go to edit sceen 1`] = `
Object {
  "pushState": Array [
    Array [
      null,
      "edit",
      "/list/2",
    ],
  ],
  "router": Object {
    "current": "edit",
    "params": Object {
      "id": "2",
    },
    "screens": Array [
      Object {
        "name": "home",
        "path": "/",
      },
      Object {
        "name": "list",
        "path": "/list",
      },
      Object {
        "name": "edit",
        "path": "/list/:id",
      },
    ],
  },
}
`;

exports[`k-mst-router should init with a simple router 1`] = `
Object {
  "current": null,
  "params": null,
  "screens": Array [
    Object {
      "name": "home",
      "path": "/",
    },
  ],
}
`;

exports[`k-mst-router should init with siblings routes 1`] = `
Object {
  "current": null,
  "params": null,
  "screens": Array [
    Object {
      "name": "home",
      "path": "/",
    },
    Object {
      "name": "login",
      "path": "/login",
    },
    Object {
      "name": "list",
      "path": "/list",
    },
  ],
}
`;
